tests/run/main-annotation-homemade-annot-3.scala

```diff
@@ -0,0 +1,23 @@
+import scala.annotation.MainAnnotation
+
+@mainNoArgs def foo() = println("Hello world!")
+
+object Test:
+  def main(args: Array[String]) =
+    val clazz = Class.forName("foo")
+    val method = clazz.getMethod("main", classOf[Array[String]])
+    method.invoke(null, Array[String]())
+end Test
+
+class mainNoArgs extends MainAnnotation:
+  override type ArgumentParser[T] = util.CommandLineParser.FromString[T]
+  override type MainResultType = Any
+
+  override def command(args: Array[String], commandName: String, docComment: String) =
+    new MainAnnotation.Command[ArgumentParser, MainResultType]:
+      override def argGetter[T](paramInfos: MainAnnotation.ParameterInfos[T])(using p: ArgumentParser[T]): () => T = ???
+
+      override def varargGetter[T](paramInfos: MainAnnotation.ParameterInfos[T])(using p: ArgumentParser[T]): () => Seq[T] = ???
+
+      override def run(f: => MainResultType): Unit = f
+  end command
```

Comment:
We can check statically that the user did not add arguments using the following macro:
`tests/neg-macros/main-annotation-homemade-annot-4/Macro_1.scala`
```scala
import scala.quoted.*

def retainedError(msg: String)(using Quotes): Expr[Nothing] =
  import quotes.reflect.*
  // hacky condition, this should probably be added to the reflection API
  if Symbol.spliceOwner.owner.fullName.endsWith("$retainedBody") then
    '{ throw new Error(${Expr(msg)}) }
  else
    report.errorAndAbort(msg, paramInfos)
```
`tests/neg-macros/main-annotation-homemade-annot-4/Test_2.scala`
```scala
import scala.annotation.MainAnnotation

class mainNoArgs extends MainAnnotation:
  override type ArgumentParser[T] = util.CommandLineParser.FromString[T]
  override type MainResultType = Any

  override def command(args: Array[String], commandName: String, docComment: String): NoArgsCommand =
    new NoArgsCommand

  final class NoArgsCommand extends MainAnnotation.Command[ArgumentParser, MainResultType]:
    inline override def argGetter[T](paramInfos: MainAnnotation.ParameterInfos[T])(using p: ArgumentParser[T]): () => T =
      ${ retainedError("@mainNoArgs does not allow arguments") }

    inline override def varargGetter[T](paramInfos: MainAnnotation.ParameterInfos[T])(using p: ArgumentParser[T]): () => Seq[T] =
      ${ retainedError("@mainNoArgs does not allow arguments") }

    override def run(f: => MainResultType): Unit = f

@mainNoArgs def foo() = () // ok

@mainNoArgs def bar(i: Int) = () // error

@mainNoArgs def baz(i: String*) = () // error
```

Reply:
We should add this test case

Reply:
> ```scala
>   else
>     report.errorAndAbort(msg, paramInfos)
> ```

What is paramInfos here?